external-crates/move/move-compiler/src/parser/syntax.rs

```diff
@@ -2183,6 +2176,55 @@ fn parse_postfix_ability_declarations(
     Ok(())
 }
 
+fn check_struct_visibility(visibility: Option<Visibility>, context: &mut Context) {
+    let current_package = context.package_name;
+    match visibility {
+        Some(Visibility::Public(loc))
+            if context.env.check_feature(
+                &FeatureGate::PublicStructTypeVisibility,
+                current_package,
+                loc,
+            ) => {}
+        Some(vis)
+            if context
+                .env
+                .supports_feature(current_package, &FeatureGate::PublicStructTypeVisibility) =>
+        {
+            let msg = format!(
+                "Invalid struct declaration. Structs can only have have the '{}' visibility modifier",
```

Comment:
I'd probably say something like `'{vis}' struct declarations are not yet supported`